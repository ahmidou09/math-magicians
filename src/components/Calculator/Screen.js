import React from 'react';
import PropTypes from 'prop-types';

const Screen = ({ next, total }) => (
  <div className="screen">{next || total || '0'}</div>
);

Screen.propTypes = {
  next: PropTypes.string,
  total: PropTypes.string,
};

Screen.defaultProps = {
  next: '',
  total: '',
};

export default Screen;
