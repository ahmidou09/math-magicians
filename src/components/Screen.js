const Screen = () => <div className="screen">0</div>;
export default Screen;
