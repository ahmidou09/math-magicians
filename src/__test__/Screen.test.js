/* eslint-disable linebreak-style */
import React from 'react';
import { render } from '@testing-library/react';
import Screen from '../components/Calculator/Screen';

test('renders Screen component with next value', () => {
  const { container } = render(<Screen next="42" />);
  expect(container).toMatchSnapshot();
});

test('renders Screen component with total value', () => {
  const { container } = render(<Screen total="123" />);
  expect(container).toMatchSnapshot();
});

test('renders Screen component with default value', () => {
  const { container } = render(<Screen />);
  expect(container).toMatchSnapshot();
});
