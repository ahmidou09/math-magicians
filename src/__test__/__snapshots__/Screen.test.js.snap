// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Screen component with default value 1`] = `
<div>
  <div
    class="screen"
    data-testid="screen"
  >
    0
  </div>
</div>
`;

exports[`renders Screen component with next value 1`] = `
<div>
  <div
    class="screen"
    data-testid="screen"
  >
    42
  </div>
</div>
`;

exports[`renders Screen component with total value 1`] = `
<div>
  <div
    class="screen"
    data-testid="screen"
  >
    123
  </div>
</div>
`;
